os/win/system.cpp

```diff
@@ -155,9 +159,24 @@ SystemWin::SystemWin()
 
 SystemWin::~SystemWin()
 {
+  if (m_appMode == AppMode::GUI) {
+    OleUninitialize();
+  }
   destroyInstance();
 }
 
+void SystemWin::setAppMode(AppMode appMode)
+{
+  m_appMode = appMode;
+  if (m_appMode == AppMode::GUI) {
+    auto result = OleInitialize(nullptr);
+    if (result != S_OK && result != S_FALSE) {
+      LOG(LogLevel::ERROR, "WIN: Could not initialize OLE (%d)", result);
+      return;
```

Comment:
No need for this return as we'll return anyway and we should keep running if the `OleInitialize` fails.
```suggestion
```